src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -13,6 +13,48 @@ namespace Microsoft.PowerFx.Functions
 {
     internal static partial class Library
     {
+        public static FormulaValue LookUp(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, FormulaValue[] args)
+        {
+            // Streaming 
+            var arg0 = (TableValue)args[0];
+            var arg1 = (LambdaFormulaValue)args[1];
+            var arg2 = (LambdaFormulaValue)(args.Length > 2 ? args[2] : null);
+
+            if (args.Length > 3)
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = "LookUp() only supports two predicates",
+                    Span = irContext.SourceContext,
+                    Kind = ErrorKind.Validation
+                });
+            }
+
+            var row = LazyFilter(runner, symbolContext, arg0.Rows, arg1).FirstOrDefault();
+
+            if (row != null)
+            {
+                if (args.Length == 2)
+                {
+                    return row?.ToFormulaValue() ?? new BlankValue(irContext);
```

Comment:
>row? [](http://example.com/codeflow?start=27&length=4)

No need for the `?` here since you already know that `row` is not null (4 lines above)